services/admin/src/server.pins.test.ts

```diff
@@ -0,0 +1,41 @@
+import { Application } from 'express';
+import request from 'supertest';
+import { dirSync } from 'tmp';
+
+import { generatePin } from '@votingworks/utils';
+import { mockOf } from '@votingworks/test-utils';
+
+import { buildApp } from './server';
+import { createWorkspace, Workspace } from './util/workspace';
+
+jest.mock('@votingworks/utils', (): typeof import('@votingworks/utils') => {
+  return {
+    ...jest.requireActual('@votingworks/utils'),
+    generatePin: jest.fn(),
+  };
+});
+
+let app: Application;
+let workspace: Workspace;
+
+beforeEach(() => {
+  mockOf(generatePin).mockReset();
+
+  workspace = createWorkspace(dirSync().name);
+  app = buildApp({ workspace });
+});
+
+afterAll(() => {
+  jest.restoreAllMocks();
+});
+
+describe('POST admin/pins/generate-card-pin', () => {
```

Comment:
Doesn't hurt to have it, but we typically don't use `describe` unless we really want to separate things within a file. Even then, it's probably best to split into separate files.